proto/noble/dollar/vaults/v2/query.proto

```diff
@@ -0,0 +1,589 @@
+message QueryInflightFundRequest {
+  uint32 route_id = 1;
+}
+
+// QueryInflightFundResponse returns an in-flight operation
+message QueryInflightFundResponse {
+  InflightFund fund = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryInflightFundsUserRequest requests all in-flight operations for a user
+message QueryInflightFundsUserRequest {
+  string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+// QueryInflightFundsUserResponse returns all in-flight operations for a user
+message QueryInflightFundsUserResponse {
+  repeated InflightFund funds = 1 [(gogoproto.nullable) = false];
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryCrossChainSnapshotRequest requests cross-chain snapshot for a vault
+message QueryCrossChainSnapshotRequest {
+  int64 timestamp = 1; // Optional: specific timestamp, 0 for latest
+}
+
+// QueryCrossChainSnapshotResponse returns cross-chain snapshot
+message QueryCrossChainSnapshotResponse {
+  CrossChainPositionSnapshot snapshot = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryStaleInflightAlertsRequest requests stale inflight alerts
+message QueryStaleInflightAlertsRequest {
+  uint32 route_id = 1; // Optional: filter by route
+  string address = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // Optional: filter by user
+  cosmos.base.query.v1beta1.PageRequest pagination = 3;
+}
+
+// QueryStaleInflightAlertsResponse returns stale inflight alerts
+message QueryStaleInflightAlertsResponse {
+  repeated StaleInflightAlertWithDetails alerts = 1 [(gogoproto.nullable) = false];
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// StaleInflightAlertWithDetails includes additional context with the alert
+message StaleInflightAlertWithDetails {
+  StaleInflightAlert alert = 1 [(gogoproto.nullable) = false];
+  CrossChainRoute route = 2 [(gogoproto.nullable) = false];
+  InflightFund fund = 3 [(gogoproto.nullable) = false];
+}
+
+// ========== Spec-aligned Query Messages ==========
+
+// NAV (spec-aligned)
+message QueryNAVRequest {}
+
+message NAVBreakdownView {
+  string local = 1;
+  string remote_positions = 2;
+  string inflight = 3;
+  string liabilities = 4;
+  string total = 5;
+}
+
+message QueryNAVResponse {
+  string nav = 1; // Total Net Asset Value ($USDN)
+  string yield_rate = 2; // APY as Dec string
+  google.protobuf.Timestamp last_update = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true
+  ];
+  string total_deposits = 4;
+  string total_accrued_yield = 5;
+  string local_assets = 6;
+  string remote_positions_value = 7;
+  string inflight_funds_value = 8;
+  string pending_withdrawals = 9;
+  NAVBreakdownView nav_breakdown = 10;
+}
+
+// Withdrawal queue views
+message QueryWithdrawalQueueRequest {
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+
+message WithdrawalQueueItem {
+  string request_id = 1;
```

Comment:
Is there any reason why it should be a `string` instead of a `uint64` as in the [specs](https://github.com/noble-assets/dollar/pull/56/files#diff-dd437411e5e4b0775b33bbebed5a4657f348bcedf4cf51af27f905ec2dca5671R138)?